cloud/cloud_intent.go

```diff
@@ -45,5 +46,74 @@ func NewIntent(d distro.Distro, instanceName, provider string, options HostOptio
 		intentHost.ProvisionOptions = options.ProvisionOptions
 	}
 	return intentHost
+}
+
+// GenerateContainerHostIntents generates container intent documents by going
+// through available parents and packing on the parents with longest expected
+// finish time
+func GenerateContainerHostIntents(d distro.Distro, newContainersNeeded int, hostOptions HostOptions) ([]host.Host, error) {
```

Comment:
These should go in the model/host package. They aren't really cloud provider-related (ie. we would use this same logic regardless of if the container provider were Docker) and also we don't want the scheduler depending on the cloud package. cloud.NewIntent can move into the host package as well